h2/config.py

```diff
@@ -25,6 +26,21 @@ def __set__(self, instance, value):
         setattr(instance, self.attr_name, value)
 
 
+class DummyLogger(logging.Logger):
+    """
+    An Logger object that does not actual logging, hence a DummyLogger.
+
+    For the class the log operation is merely a no-op.  The intent is to avoid
+    conditionals being sprinkled throughout the hyper-h2 code for calls to
+    logging functions when no logger is passed into the corresponding object.
+    """
+    def _log(self, *vargs, **kwargs):
+        """
+        No-op logging.
+        """
+        pass
```

Comment:
This was my suggestion. I struggled to come up with an obviously better alternative:

* We could use mock, but I don’t think we need that dependency.
* Anything involving `__getattr__` magic. Just say no.
* Patch every public method on the `Logger` class (or create an entirely new class that mimics its interface). Quite verbose.

But if you have an alternative suggestion, I’d be all ears. :-)